crates/lakekeeper/src/catalog/namespace.rs

```diff
@@ -72,15 +73,31 @@ impl<C: Catalog, A: Authorizer + Clone, S: SecretStore>
             .await?;
         let mut t = C::Transaction::begin_read(state.v1_state.catalog.clone()).await?;
 
+        let mut can_list_everything = false;
         if let Some(parent) = parent {
             let namespace_id = C::namespace_to_id(warehouse_id, parent, t.transaction()).await; // Cannot fail before authz
+            let maybe_namespace_id = match namespace_id {
+                Ok(Some(id)) => Some(id),
+                _ => None,
+            };
+
```

Comment:
This is not not needed and could be achieved more efficiently using `.ok.flatten`.
Instead, lets do the following:

```rust
            let namespace_id = authorizer
                .require_namespace_action(
                    &request_metadata,
                    namespace_id,
                    CatalogNamespaceAction::CanListNamespaces,
                )
                .await?;

            can_list_everything = authorizer
                .is_allowed_namespace_action(
                    &request_metadata,
                    namespace_id,
                    CatalogNamespaceAction::CanListEverythingInNamespace,
                )
                .await?;
```